### PR TITLE
[PCX-2102] Example Objective-C application

### PR DESCRIPTION
**Pull request by Sorix**

### CODE REVIEW

**Review by caiomello — DISMISSED** [review]
😱 I forgot how terrible Objective-C looks (and I actually liked it before Swift!).

I'm pointing out a few tiny things I noticed (feel free to ignore), but this is good to go IMO ✅

File: examplecheckout-objc/Sources/AppDelegate.m [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@implementation AppDelegate
+
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // Override point for customization after application launch.
+    return YES;
+}
+
+
```
[the remark this answers is not in the document]
Comment (Sorix) [reply examplecheckout-objc/Sources/AppDelegate.m]: Resolved in 305d96b

File: examplecheckout-objc/Sources/ViewController.m [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#import "ViewController.h"
+#import <PayoneerCheckout/PayoneerCheckout-Swift.h>
+
+@interface ViewController ()
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
```
[the remark this answers is not in the document]
Comment (Sorix) [reply examplecheckout-objc/Sources/ViewController.m]: Resolved in 305d96b

**Review by caiomello — COMMENTED** [review]
Awesome that you managed to make it into a single project!

What are the trade-offs here? Any downsides? Potential issues?

**Comment by Sorix** [comment]
@caiomello I don't see any tradeoffs or cons in that solution comparing to separate projects. Please make an approval review if everything is okay 🙏

**Comment by Sorix** [comment]
> Awesome that you managed to make it into a single project!
> 
> What are the trade-offs here? Any downsides? Potential issues?

@caiomello I don't see any tradeoffs or cons in that solution comparing to separate projects. Please make an approval review if everything is okay 🙏

**Review by caiomello — APPROVED** [review]
I checked out the branch locally to see how the mixed project is set up and it looks great!

I noticed the tests only run for the Swift target. Do we have plans to add UI tests to the ObjC example app?

**Review by siebeprojects — DISMISSED** [review]

**Comment by Sorix** [comment]
> I checked out the branch locally to see how the mixed project is set up and it looks great!
> I noticed the tests only run for the Swift target. Do we have plans to add UI tests to the ObjC example app?
Yes, tests are run against Swift binary. Make same tests to run against Objective-C is not so easy task and that will be part of another ticket.

**Review by caiomello — APPROVED** [review]

**Review by siebeprojects — APPROVED** [review]

**Review by bzlat — APPROVED** [review]

**Pull request merged**